### PR TITLE
vimPlugins.remote-nvim-nvim: init at 2024-08-04

### PR DESCRIPTION
**Pull request by khaneliman**
## Things done
Created for https://github.com/nix-community/nixvim/issues/2546 and tested with https://github.com/NixOS/nixpkgs/pull/352277 to find dependencies.

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by PerchunPak** [comment]
It fails on #352277 

```
Attempting to require module: repro
Error detected while processing pre-vimrc command line:
E5108: Error executing lua ./repro.lua:2: attempt to call a nil value
stack traceback:
        ./repro.lua:2: in main chunk
        [C]: in function 'require'
        [string ":lua"]:1: in main chunk
Entering Ex mode.  Type "visual" to go to Normal mode.
:Failed to require module: repro
```

I think, we can start to add `nvimSkipModule` for new plugins

**Comment by khaneliman** [comment]
> It fails on #352277
> 
> ```
> Attempting to require module: repro
> Error detected while processing pre-vimrc command line:
> E5108: Error executing lua ./repro.lua:2: attempt to call a nil value
> stack traceback:
>         ./repro.lua:2: in main chunk
>         [C]: in function 'require'
>         [string ":lua"]:1: in main chunk
> Entering Ex mode.  Type "visual" to go to Normal mode.
> :Failed to require module: repro
> ```
> 
> I think, we can start to add `nvimSkipModule` for new plugins

I figured I'd just add the nvimSkipModule in that PR, unless we are able to merge it soon and it makes sense to just include it early.

**Comment by PerchunPak** [comment]
3 approvals is enough to merge it right now IMO

File: pkgs/applications/editors/vim/plugins/overrides.nix [hunk 1]
```diff
@@ -2234,6 +2234,14 @@ in
     nvimRequireCheck = "refactoring";
   };
 
+  remote-nvim-nvim = super.remote-nvim-nvim.overrideAttrs {
+    dependencies = with self; [
+      nui-nvim
+      plenary-nvim
+    ];
+    nvimSkipModule = "repro";
```
Comment (PerchunPak) [inline pkgs/applications/editors/vim/plugins/overrides.nix:2242]: It shouldn't be repro...

https://github.com/amitds1997/remote-nvim.nvim/blob/main/repro.lua

Comment (khaneliman) [reply pkgs/applications/editors/vim/plugins/overrides.nix:2242]: Not sure what you mean? 
image

Comment (PerchunPak) [reply pkgs/applications/editors/vim/plugins/overrides.nix:2242]: Oh, sorry, confused this with `nvimRequireCheck`

**Review by PerchunPak — APPROVED** [review]

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 360621`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.remote-nvim-nvim</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.remote-nvim-nvim</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.remote-nvim-nvim</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.remote-nvim-nvim</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

**Pull request merged**